### ISSUE TITLE
Support switches/joystick/touch

### ISSUE THREAD

**Issue by kmcnaught** [OptiKey/OptiKey#556]
Joystick input for switches.
Switch scanning.
Touch input – always available (+ new setting to disable if not wanted).

**Comment by dreua** [comment]
Is this a duplicate of  #412 or do you mean something else?

**Comment by JuliusSweetland** [comment]
@dreua Yes this overlaps, but this issue includes touch support and general switch support. We'll keep both issues until we can disambiguate them.

**Comment by JuliusSweetland** [comment]
Switch input should have multiple selection/progression methods:
1.Auto scanning with switch selection. Touching the screen should be a valid switch input method.
2.Two switches: one to advance, the other to select.
3.One switch - press to advance, hold to select.

**Comment by tqphan** [comment]
If you need inspirations on how to implement switch access for mouse movements, look at MacOS's implementation. I thought it was done very well.

**Comment by kmcnaught** [comment]
Renamed to reduce scope to switch scanning. Joysticks are now in, and #23 covers touch

**Comment by alexandre-mbm** [comment]
Fake keyboards built from arduino-like boards, what are?

**Comment by kmcnaught** [comment]
@alexandre-mbm an arduino should be able to emulate a keyboard or a joystick, in which case it could be used as a trigger with Optikey. @annakirkpatrick has some experience of this. Do you have an arduino setup you are wanting to use?

**Comment by alexandre-mbm** [comment]
@kmcnaught, I tried an arduino so years ago. I know that works. But I am curious if commercial switches would be to classified in this way. A more elegant labeling like device drive.
